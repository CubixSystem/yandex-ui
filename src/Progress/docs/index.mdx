---
id: Progress
path: /components/progress/examples

title: Progress
description: Компонент для создания полосы прогресса.

links:
 - label: source
   url: https://github.yandex-team.ru/search-interfaces/frontend/tree/master/packages/lego-components/src/Progress
 - label: startrek
   url: https://search.yandex-team.ru/stsearch?text=Progress.ts&facet.queue=ISL&facet.type=bug&facet.status=128
 - label: figma
   url: https://www.figma.com/file/CaoHFEO0w6PaE8cVaYFDoq8i/L-E-G-O?node-id=0%3A1

tabs:
 - Examples
 - Usage
 - Playground
 - Props
---
