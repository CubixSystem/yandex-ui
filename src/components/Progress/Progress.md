# Progress

<a href='https://github.yandex-team.ru/search-interfaces/frontend/tree/master/packages/lego-components/src/components/Progress' target='_blank'><img src='https://badger.yandex-team.ru/custom/[Исходники]/[Github
][green]/badge.svg' /></a> <a href='https://search.yandex-team.ru/stsearch?text=Progress.ts&facet.queue=ISL&facet.type=bug&facet.status=128' target='_blank'><img src='https://badger.yandex-team.ru/custom/[Известные проблемы]/[Startrek][blue]/badge.svg' /></a>

<!-- description:start -->
Компонент для создания полосы прогресса.
<!-- description:end -->

## Пример использования

Конфигурация темы на уровне проекта:

```ts
// src/lib/theme.ts
import { configureRootTheme } from '@yandex-lego/components/Theme'
import { theme } from '@yandex-lego/components/Theme/presets/default'

configureRootTheme({ theme })
```

Использование компонента:

```ts
// src/App.ts
import React from 'react'
import { Progress } from '@yandex-lego/components/Progress/desktop'

const App = () => {
  return <Progress timing="linear" value={50} maxValue={100}/>
}
```

## Свойства

<!-- props:start -->
| Свойство   | Тип                         | По умолчанию | Описание                                                 |
| ---------- | --------------------------- | ------------ | -------------------------------------------------------- |
| maxValue?  | `number`                    | `1`          | Максимальное допустимое значение прогресс бара           |
| value?     | `number`                    | `0`          | Доля загрузки прогресс бара от 0 до maxValue             |
| timing?    | `"linear"`                  | —            | Способ CSS-анимации при изменении ширины полосы загрузки |
| innerRef?  | `RefObject<HTMLDivElement>` | —            | Ссылка на корневой DOM-элемент компонента                |
| style?     | `CSSProperties`             | —            | Пользовательские стили                                   |
| className? | `string`                    | —            | Дополнительный класс                                     |
<!-- props:end -->
