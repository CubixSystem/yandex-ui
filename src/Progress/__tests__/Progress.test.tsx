import React, { createRef } from 'react';

import { createClientRender, screen } from '../../internal/testing';
import { Progress } from '../Progress';

describe('Progress', () => {
    const render = createClientRender();

    test('должен вернуть полный шаблон компонента (snapshot)', () => {
        const { container } = render(<Progress className="mix" />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('должен быть установлен корректный displayName', () => {
        expect(Progress.displayName).toBe('Progress');
    });

    test('должен устанавливать ссылку на корневой DOM элемент', async () => {
        const innerRef = createRef<HTMLDivElement>();
        render(<Progress innerRef={innerRef} />);

        expect(innerRef.current).not.toBe(null);
    });

    test('должен устанавливать и изменять value через props', () => {
        const { setProps } = render(<Progress value={0.5} data-testid="element" />);

        expect(screen.getByTestId('element').firstChild).toHaveStyle({
            width: '50.00%',
        });

        setProps({ value: 1 });

        expect(screen.getByTestId('element').firstChild).toHaveStyle({
            width: '100%',
        });
    });

    test('должен устанавливать value учитывая maxValue', () => {
        render(<Progress value={50} maxValue={150} data-testid="element" />);

        expect(screen.getByTestId('element').firstChild).toHaveStyle({
            width: '33.33%',
        });
    });

    test('должен устанавливать и изменять timing через props', () => {
        render(<Progress timing="linear" data-testid="element" />);

        expect(screen.getByTestId('element')).toHaveClass('Progress_timing_linear');
    });
});
