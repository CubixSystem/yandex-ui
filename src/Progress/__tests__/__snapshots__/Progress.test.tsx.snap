// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress должен вернуть полный шаблон компонента (snapshot) 1`] = `
<div
  class="Progress mix"
>
  <div
    class="Progress-Inner"
    style="width: 0px;"
  />
</div>
`;
